### ISSUE TITLE
[Bug]: Health attribute from items is ignored

### ISSUE THREAD

**Issue by ericdm47** [magemonkeystudio/fabled#473]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Have you tried to reproduce this without OP?

- [x] I have tested without OP

### Current Behavior

base health is 20 without increasing your health with attributes. 
20 health not increased by attributes.
However, the item increased health.
When HP is restored with a health potion, etc., it is restored to basic HP.

![](https://cdn.discordapp.com/attachments/1103980100367691787/1103984036227059723/image.png)

### Expected Behavior

proskillapi doesn't seem to recognize item nbt

### Steps To Reproduce

1. Give the item maximum health with the nbt tag.
2. Wear it.
3. When recovering, max health is not applied and returns to basic health

### Environment

```markdown
- Server Software: paper 1.19.4 - 519
- ProSkillAPI Version: 1.1.8 R0.27-20230505
- ProMCCore Version: 1.0.4 R0.7 20230504
- Plugins: mmoitem, mimic ect.,
```

### latest.log

don't need a .log file because these are errors that don't show up in the log.

**Comment by Travja** [comment]
I'm unable to reproduce this.

**Comment by Sentropic** [comment]
This is most likely due to some of your other plugins. Send your full plugin list and try to isolate the problematic one so that we can work on fixing it.

**Comment by xMystik** [comment]
Im having the same issue for quite some time now, mostly I was being able to reproduce it by using /heal or being healed by skills, and then if I have armor that gives extra health and change the holding item the health resets back to the one that I had before equipping the armor.

**Comment by Travja** [comment]
@xMystik I finally got my test setup to glitch into a state that I was able to reproduce this (was actually working on magemonkeystudio/divinity#231 and got some dangling attribute modifiers applied to my player). Prior to this fix, I had been calculating the "modified max health" of the player by taking their class base hp and applying item bonuses to their HP. If the player's health was above that modified value, the plugin would then bring their health down to match what should have been the max health anyway. This system just wasn't taking into account the possibility that other plugins might add a bonus to those without it being registered on an item.
